extensions/sql-database-projects/src/dialogs/utils.ts

```diff
@@ -25,11 +26,45 @@ export function getConnectionName(connection: any): string {
 	return connectionName;
 }
 
+export function getAgreementDisplayText(agreementInfo: AgreementInfo): string {
+	// the agreement template will have {index} as placeholder for hyperlinks
+	// this method will get the display text after replacing the placeholders
+	let text = agreementInfo.template;
+	text = text.replace(`{0}`, agreementInfo.link!.text);
```

Comment:
You should be passing the text in as a parameter to the localize function instead.